pyccel/stdlib/ndarrays/ndarrays.c

```diff
@@ -0,0 +1,124 @@
+#include "ndarrays.h"
+
+/* 
+** allocation
+*/
+
+t_ndarray *init_array(char *temp, int nd, int *shape, int type)
+{
+	t_ndarray *a;
+		
+	a = malloc(sizeof(t_ndarray));
+	a->type = type;
+	a->nd = nd;
+	a->shape = malloc(a->nd * sizeof(int));
+	a->lenght = 1;
+	a->is_slice = 0;
+	for (int i = 0; i < a->nd; i++) // init the shapes
+	{
+		a->lenght *= shape[i]; 
+		a->shape[i] = shape[i];
+	}
+	a->strides = malloc(nd * sizeof(int));
+	for (int i = 0; i < a->nd; i++)
+	{
+		a->strides[i] = 1;
+		for (int j = i + 1; j < a->nd; j++)
+			a->strides[i] *= a->shape[j];
+	}
+	a->data = malloc(sizeof(t_ndarray_type));
+	a->data->raw_data = calloc(a->lenght , a->type);
```

Comment:
@yguclu Do you think we should be using `calloc` or `malloc` here? I would expect `malloc` to be used for `np.empty` and `calloc` to be used for `np.zeros` as per the numpy description of `empty`:
> `empty`, unlike `zeros`, does not set the array values to zero,
> and may therefore be marginally faster.  On the other hand, it requires
> the user to manually set all the values in the array, and should be
> used with caution.

Reply:
yeah i will split init_array to have an array_empty function